sql/dd/impl/system_views/compression_dictionary.cc

```diff
@@ -0,0 +1,40 @@
+/*****************************************************************************
+
+Copyright (c) 2018, Percona Inc. All Rights Reserved.
+
+This program is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Suite 500, Boston, MA 02110-1335 USA
+
+*****************************************************************************/
+
+#include "sql/dd/impl/system_views/compression_dictionary.h"
+
+namespace dd {
+namespace system_views {
+
+const Compression_dictionary &Compression_dictionary::instance() {
+  static Compression_dictionary *s_instance = new Compression_dictionary();
```

Comment:
I followed the other view creations. For example:

For example:
```
const Schemata &Schemata::instance() {
  static Schemata *s_instance = new Schemata();
  return *s_instance;
}
```
May be the memory is freed on shutdown

Reply:
sql/dd/impl/system_views/schemata.cc

Reply:
This only makes difference if `Schemata` has overloaded operator `new`. Otherwise, statically allocated object is better from the memory leaks detection point of view.

Reply:
Ok, Let me check how the object is freed in current way. I'm fine with either way.

Reply:
There is no memory leak and the free is done by destructor

#0  __GI___libc_free (mem=0x55555bc27420) at malloc.c:3086
#1  0x00005555591036fc in dd::system_views::Schemata::~Schemata (this=0x55555bc27420, __in_chrg=<optimized out>, __vtt_parm=<optimized out>)
    at /home/satya/WORK/PS-8.0-4805/sql/dd/impl/system_views/schemata.h:37
#2  0x0000555558fa3122 in dd::Entity_element<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, Stateless_allocator<char, dd::String_type_alloc, My_free_functor> > const, std::__cxx11::basic_string<char, std::char_traits<char>, Stateless_allocator<char, dd::String_type_alloc, My_free_functor> > const>, dd::system_views::System_view const, dd::System_views::Types, &dd::System_views::type_name, true>::~Entity_element (this=0x55555bc279b0, 
    __in_chrg=<optimized out>) at /home/satya/WORK/PS-8.0-4805/sql/dd/impl/system_registry.h:76
#3  0x0000555558fa1c8a in dd::Entity_registry<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, Stateless_allocator<char, dd::String_type_alloc, My_free_functor> > const, std::__cxx11::basic_string<char, std::char_traits<char>, Stateless_allocator<char, dd::String_type_alloc, My_free_functor> > const>, dd::system_views::System_view const, dd::System_views::Types, &dd::System_views::type_name, true>::~Entity_registry (
    this=0x55555b9a0da0 <dd::System_views::instance()::s_instance>, __in_chrg=<optimized out>)
    at /home/satya/WORK/PS-8.0-4805/sql/dd/impl/system_registry.h:147
#4  0x0000555558fa0d24 in dd::System_views::~System_views (this=0x55555b9a0da0 <dd::System_views::instance()::s_instance>, __in_chrg=<optimized out>)
    at /home/satya/WORK/PS-8.0-4805/sql/dd/impl/system_registry.h:457
#5  0x00007ffff5db5041 in __run_exit_handlers (status=0, listp=0x7ffff615d718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, 
    run_dtors=run_dtors@entry=true) at exit.c:108
#6  0x00007ffff5db513a in __GI_exit (status=<optimized out>) at exit.c:139
#7  0x0000555557e45611 in mysqld_exit (exit_code=0) at /home/satya/WORK/PS-8.0-4805/sql/mysqld.cc:1919
#8  0x0000555557e55b0e in mysqld_main (argc=66, argv=0x55555ba6fb88) at /home/satya/WORK/PS-8.0-4805/sql/mysqld.cc:6721
#9  0x0000555557c776ca in main (argc=8, argv=0x7fffffff5d78) at /home/satya/WORK/PS-8.0-4805/sql/main.cc:30

Reply:
Since there is no correctness issue, I will keep upstream style